filament/backend/src/vulkan/VulkanDriver.cpp

```diff
@@ -888,12 +895,52 @@ void VulkanDriver::beginRenderPass(Handle<HwRenderTarget> rth, const RenderPassP
         }
     }
 
-    const VkCommandBuffer cmdbuffer = mContext.commands->get().cmdbuffer;
     VulkanAttachment depth = rt->getSamples() == 1 ? rt->getDepth() : rt->getMsaaDepth();
+#ifdef VULKAN_DEBUG_REAL
+    if (depth.texture) {
+        depth.texture->print();
+    }
+#endif
 
-    VulkanDepthLayout initialDepthLayout = fromVkImageLayout(depth.getLayout());
-    VulkanDepthLayout renderPassDepthLayout = VulkanDepthLayout::ATTACHMENT;
-    VulkanDepthLayout finalDepthLayout = VulkanDepthLayout::ATTACHMENT;
+    bool samplingDepthAttachment = false;
```

Comment:
could we get a comment that explains a bit what this loop does? and/or could this loop be factored out?

Reply:
added a comment.  Not sure how it could be factored out, but maybe adding a hint in the backend API (for when a texture is both an attachment and a sampler) can make this loop unnecessary.

Reply:
`createTextureR` takes a `TextureUsage` parameter that should tell you that.